t/14-isotovideo.t

```diff
@@ -22,27 +23,28 @@ my $pool_dir     = "$dir/pool";
 chdir $dir;
 my $cleanup = scope_guard sub { chdir $Bin; undef $dir };
 mkdir $pool_dir;
-my $log_file = path('autoinst-log.txt');
-my $log      = '';
 
 sub isotovideo {
     my (%args) = @_;
     $args{default_opts} //= 'backend=null';
     $args{opts}         //= '';
     $args{exit_code}    //= 1;
-    my $cmd = "perl $toplevel_dir/isotovideo -d $args{default_opts} $args{opts} 2>&1 | tee autoinst-log.txt";
-    note("Starting isotovideo with: $cmd");
-    system($cmd);
-    $log = $log_file->slurp;
-    like $log, qr/\d*: EXIT $args{exit_code}/, $args{end_test_str} ? $args{end_test_str} : 'isotovideo run exited as expected';
+    my @cmd = ($^X, "$toplevel_dir/isotovideo", '-d', $args{default_opts}, split(' ', $args{opts}));
+    note("Starting isotovideo with: @cmd");
+    capture [$args{exit_code}], @cmd;
 }
 
+subtest 'get the version number' => sub {
+    # Make sure we're in a folder we can't write to, no base_state.json should be created here
+    chdir('/');
+    combined_like { system $^X, "$toplevel_dir/isotovideo", '--version' } qr/Current version is.+\[interface v20\]/, 'version printed';
+    ok(!-e bmwqemu::STATE_FILE, 'no state file was written');
```

Comment:
I do not understand this test code clearly. is it used to prove that we don't write any message to `STATE_FILE ` file?  I guess we also need to check $fatal_error will not be filed with messages when running `--version`

Reply:
Yes. If `$fatal_error` is set, the file is written. So I'm checking that it's not written. Which means there's no error.